internal/provider/resource_gitlab_personal_access_tokens.go

```diff
@@ -0,0 +1,265 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var validPersonalAccessTokenScopes = []string{
+	"api",
+	"read_user",
+	"read_api",
+	"read_repository",
+	"write_repository",
+	"read_registry",
+	"write_registry",
+	"sudo",
```

Comment:
Where do you see "read_registry", "write_registry", and "sudo" documented? They aren't listed here: https://docs.gitlab.com/ee/api/users.html#create-a-personal-access-token

Reply:
Sudo only appears when you are an admin on a self-hosted GitLab instance and read/write registry are there for all users on SAAS and self-hosted.  It appears that GitLab's documentation is out of date for their API as the UI docs are up to date.
https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#personal-access-token-scopes

Reply:
So I found something really off with the underlying GitLab API. In the API docs it leaves off a few scopes (read_registry, write_registry, sudo). Of these missing from the API documentation only `sudo` can be set out of those scopes via the API. In the UI the end user is fully capable of adding `read_registry` and `write_registry` but errors out with `400 {error: scopes does not have a valid value}` if done via the provider/API for . I will try to find a ticket in GitLab to see if anyone else has noticed this inconsistency or if the API update is planned for a future release but is only UI settable for now

Reply:
@nagyv or @DonNicoJs I was unable to find anything in the GitLab issue tracker can you see on your end if anyone is aware of the API inconsistency for user PAT scopes?

Reply:
Ok cool. This is the last remaining thing before I can approve. 😄

Reply:
As to not lose sight of this difference in the API vs UI scopes for the future how do you want to capture the difference?

Reply:
@Shocktrooper Hello, GitLab team member here. I didn't work on this API, but I took a look at the code and it validates the scopes using [`Gitlab::Auth#all_available_scopes`](https://gitlab.com/gitlab-org/gitlab/-/blob/25be580b697e5697df4288e5b5064bc1a3e3a480/lib/gitlab/auth.rb#L386). The `read_registry` and `write_registry` scopes are only available if the container registry is enabled on the GitLab instance. Since this API requires administrator, I assume you are testing this with a self-hosted instance. Does it have container registry enabled?

I've also opened https://gitlab.com/gitlab-org/gitlab/-/merge_requests/84272 to add the missing scopes to the documentation. Thanks for your feedback!

Reply:
So turns out it did not work and I needed to map a gitlab.rb file in. I had a false positive because of the volume mounts persisting between tests